lms/djangoapps/certificates/views/webview.py

```diff
@@ -252,7 +253,13 @@ def _update_course_context(request, context, course, course_key, platform_name):
     if CertificateGenerationCourseSetting.is_language_specific_templates_enabled_for_course(course_key):
         fields = ['start', 'end', 'max_effort', 'content_language']
         course_run_data = get_course_run_details(course_key, fields)
-        context.update(course_run_data)
+        if course_run_data['start'] and course_run_data['end'] and course_run_data['max_effort']:
+            # Calculate duration of the course run in weeks, multiplied by max_effort for total Hours of Effort
+            start = parser.parse(course_run_data['start'])
+            end = parser.parse(course_run_data['end'])
+            max_effort = int(course_run_data['max_effort'])
+            context['hours_of_effort'] = ((end - start).days / 7) * max_effort
```

Comment:
This new code is great.  I would recommend that we wrap all of this in a `try..except` block just in case so we don't prevent a learner from seeing their cert just because something goofy happened with our hours of effort field calculations.

Reply:
Makes sense. On it!

Reply:
and log any errors if they're thrown